api/src/routes/user.ts

```diff
@@ -581,3 +582,44 @@ export const userGetRoutes: FastifyPluginCallbackTypebox = (
 
   done();
 };
+
+/**
+ * Plugin containing public GET routes for user account management. They are kept
+ * separate because they do not require CSRF protection or authorization.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done Callback to signal that the logic has completed.
+ */
+export const userPublicGetRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.get(
+    '/api/users/exists',
+    {
+      schema: schemas.userExists,
+      attachValidation: true
+    },
+    async (req, reply) => {
+      if (req.validationError) {
+        void reply.code(400);
+        return await reply.send({ exists: true });
+      }
```

Comment:
The current api:
a) "allows" check by `email`
b) Returns `false` if the validation fails

I prefer this PR's approach, but, ideally, would expect some other response for a validation error.

Reply:
Sorry for the late reply @ShaunSHamilton, @moT01.

> > " check by email
> 
> Yea, I noticed this. Do we want to support that?

Nope. The client doesn't use it, so there's no need to migrate it.

> b) Returns false if the validation fails

It does, but fortunately the client also validates before sending so it never runs into the bug.

I agree that we should send something more useful, but the client only understands `{ exists: boolean }` and I don't want to mess with it yet. I will add a TODO, though, since this isn't what we want post-MVP.